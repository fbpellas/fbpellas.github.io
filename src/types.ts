export interface Pronunciation {
  phoneme: string;
  graphemes: string;
  examples: string;
  audioPhoneme: string;
  audioExamples: string;
}

export interface Breadcrumb {
  anchor: string;
  title: string;
}

export enum QuizIndex {
  OddPhonemeOut = 0,
  PhoneticSpelling = 1,
  SameWordsDifferentStress = 2,
  WhereIsTheStress = 3,
  GuessThePattern = 4,
  ShoppingForAPresent = 5
}
