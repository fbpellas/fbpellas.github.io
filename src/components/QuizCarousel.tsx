import * as React from 'react';
import { Carousel } from 'react-bootstrap';
import { BASE_PATH_IMG } from '../constants';
import { QuizIndex } from '../types';

export interface QuizCarouselProps {
  indexCarousel: QuizIndex;
  setIndexCarousel: React.Dispatch<React.SetStateAction<QuizIndex>>;
}

const QuizCarousel: React.FC<QuizCarouselProps> = (props) => {
  const { indexCarousel, setIndexCarousel } = props;

  const handleSelect = (selectedIndex: number) => {
    setIndexCarousel(selectedIndex);
  };

  return (
    <Carousel activeIndex={indexCarousel} onSelect={handleSelect} interval={null} className="carousel-custom">
      <Carousel.Item>
        <img className="d-block w-100" src={`${BASE_PATH_IMG}quiz/ODD PHON.png`} alt="Odd Phoneme Out" />
      </Carousel.Item>
      <Carousel.Item>
        <img className="d-block w-100" src={`${BASE_PATH_IMG}quiz/PHONETIC SPELL.png`} alt="Phonetic Spelling" />
      </Carousel.Item>
      <Carousel.Item>
        <img className="d-block w-100" src={`${BASE_PATH_IMG}quiz/SAME WORDS.png`} alt="Same Words, Different Stress" />
      </Carousel.Item>
      <Carousel.Item>
        <img
          className="d-block w-100"
          src={`${BASE_PATH_IMG}quiz/WHERES THE STRESS bis.png`}
          alt="Whereâ€™s the Stress?"
        />
      </Carousel.Item>
      <Carousel.Item>
        <img className="d-block w-100" src={`${BASE_PATH_IMG}quiz/GUESS THE PATTERN.png`} alt="Guess the Pattern" />
      </Carousel.Item>
      <Carousel.Item>
        <img className="d-block w-100" src={`${BASE_PATH_IMG}quiz/SHOPPING.png`} alt="Shopping for a Present" />
      </Carousel.Item>
    </Carousel>
  );
};

export { QuizCarousel };
