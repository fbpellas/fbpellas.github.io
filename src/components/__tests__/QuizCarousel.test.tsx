import React from 'react';
import { render } from '@testing-library/react';
import { QuizCarousel } from '../QuizCarousel';
import { QuizIndex } from '../../types';

test('QuizCarousel', () => {
  const fn = jest.fn();
  const { container } = render(<QuizCarousel indexCarousel={QuizIndex.GuessThePattern} setIndexCarousel={fn} />);
  expect(container).toMatchSnapshot();
});
