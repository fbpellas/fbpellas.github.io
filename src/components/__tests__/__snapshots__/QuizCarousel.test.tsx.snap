// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuizCarousel 1`] = `
<div>
  <div
    class="carousel-custom carousel slide"
  >
    <ol
      class="carousel-indicators"
    >
      <li />
      <li />
      <li />
      <li />
      <li
        class="active"
      />
      <li />
    </ol>
    <div
      class="carousel-inner"
    >
      <div
        class="carousel-item"
      >
        <img
          alt="Odd Phoneme Out"
          class="d-block w-100"
          src="https://raw.githubusercontent.com/fbpellas/fbpellas.github.io/main/public/img/quiz/ODD PHON.png"
        />
      </div>
      <div
        class="carousel-item"
      >
        <img
          alt="Phonetic Spelling"
          class="d-block w-100"
          src="https://raw.githubusercontent.com/fbpellas/fbpellas.github.io/main/public/img/quiz/PHONETIC SPELL.png"
        />
      </div>
      <div
        class="carousel-item"
      >
        <img
          alt="Same Words, Different Stress"
          class="d-block w-100"
          src="https://raw.githubusercontent.com/fbpellas/fbpellas.github.io/main/public/img/quiz/SAME WORDS.png"
        />
      </div>
      <div
        class="carousel-item"
      >
        <img
          alt="Whereâ€™s the Stress?"
          class="d-block w-100"
          src="https://raw.githubusercontent.com/fbpellas/fbpellas.github.io/main/public/img/quiz/WHERES THE STRESS bis.png"
        />
      </div>
      <div
        class="active carousel-item"
      >
        <img
          alt="Guess the Pattern"
          class="d-block w-100"
          src="https://raw.githubusercontent.com/fbpellas/fbpellas.github.io/main/public/img/quiz/GUESS THE PATTERN.png"
        />
      </div>
      <div
        class="carousel-item"
      >
        <img
          alt="Shopping for a Present"
          class="d-block w-100"
          src="https://raw.githubusercontent.com/fbpellas/fbpellas.github.io/main/public/img/quiz/SHOPPING.png"
        />
      </div>
    </div>
    <a
      class="carousel-control-prev"
      href="#"
      role="button"
    >
      <span
        aria-hidden="true"
        class="carousel-control-prev-icon"
      />
      <span
        class="sr-only"
      >
        Previous
      </span>
    </a>
    <a
      class="carousel-control-next"
      href="#"
      role="button"
    >
      <span
        aria-hidden="true"
        class="carousel-control-next-icon"
      />
      <span
        class="sr-only"
      >
        Next
      </span>
    </a>
  </div>
</div>
`;
